src/CodeGen_ARM.cpp

```diff
@@ -517,8 +517,10 @@ void CodeGen_ARM::visit(const Min *op) {
         // Use a 2-wide vector instead
         Value *undef = UndefValue::get(f32x2);
         Constant *zero = ConstantInt::get(i32_t, 0);
-        Value *a_wide = builder->CreateInsertElement(undef, codegen(op->a), zero);
-        Value *b_wide = builder->CreateInsertElement(undef, codegen(op->b), zero);
+        Value *a = codegen(op->a);
```

Comment:
We should note that this change potentially changes the LLVM IR produced. Before, it was reliably:
     ...all the stuff in codegen(op->a)...
     ...CreateInsertElement for a...
    ...all the stuff in codegen(op->b)...
    ...CreateInsertElement for b...
now it is:
     ...all the stuff in codegen(op->a)...
    ...all the stuff in codegen(op->b)...
     ...CreateInsertElement for a...
     ...CreateInsertElement for b...

This has nothing to do with argument eval order in C++ and theoretically should not matter, but the context of this change is that this sort of thing did matter in at least one instance. I'm not sure we need a style guide rule on which order to prefer, but we should be aware of what is going on.

Reply:
Agreed with the reorder. The motivation here was mainly defensive (ie, in case it was edited or replicated to replace `undef` or `zero` with a nontrivial call).

Reply:
It also sets a safer convention for surrounding code to imitate.